examples/active_replication/flux_ingress_operator.py

```diff
@@ -0,0 +1,77 @@
+from erdos.data_stream import DataStream
+from erdos.op import Op
+from erdos.utils import setup_logging
+from flux_utils import is_ack_stream, is_control_stream, is_not_ack_stream, is_not_control_stream, is_not_back_pressure
+from flux_buffer import Buffer
+import flux_utils
+import threading
+
+
+class FluxIngressOperator(Op):
+    def __init__(self,
+                 name,
+                 output_stream_names,
+                 num_replics=2,
+                 log_file_name=None):
+        super(FluxIngressOperator, self).__init__(name)
+        self._logger = setup_logging(self.name, log_file_name)
+        self._num_replicas = num_replics
+        self._output_streams = output_stream_names
+        self._input_msg_seq_num = 0
+        self.buffer = Buffer(num_replics)   # buffer to store unacknowledged tuples
+        self._status = {}
```

Comment:
Why is this a set if we assume replicas have sequence numbers as ids? It could be a list. Is this for the future when we may have an unique id for each replica?

Reply:
oh that means a dictionary. set is declared as a = set()

Reply:
Ignore my comment. Let's assume replicas can have non-sequential ids => dict works great.